src/jdk.incubator.vector/share/classes/jdk/incubator/vector/X-VectorBits.java.template

```diff
@@ -842,31 +842,44 @@ final class $vectortype$ extends $abstractvectortype$ {
             return ($vectortype$) super.toVectorTemplate();  // specialize
         }
 
+        /**
+         * Helper function for lane-wise mask conversions.
+         * This function kicks in after intrinsic failure.
+         */
+        @ForceInline
+        private final <E>
+        VectorMask<E> defaultMaskCast(AbstractSpecies<E> dsp) {
+            boolean[] maskArray = toArray();
+            // enum-switches don't optimize properly JDK-8161245
+            return (
```

Comment:
Minor syntactic quibble: you don't need the '(` and `)` surrounding the switch expressions e.g.:
```java
   return switch (dsp.laneType.switchKey) {
       case ...
   }
```